rust/main/agents/relayer/src/msg/pending_message.rs

```diff
@@ -321,17 +325,29 @@ impl PendingOperation for PendingMessage {
     }
 
     #[instrument]
-    async fn submit(&mut self) {
+    async fn submit(&mut self) -> PendingOperationResult {
         if self.submitted {
             // this message has already been submitted, possibly not by us
-            return;
+            return PendingOperationResult::Success;
         }
 
         let state = self
             .submission_data
             .clone()
             .expect("Pending message must be prepared before it can be submitted");
 
+        // To avoid spending gas on a tx that will revert, dry-run just before submitting.
+        if let Some(metadata) = self.metadata.clone() {
+            if let Err(_) = self
+                .ctx
+                .destination_mailbox
+                .process_estimate_costs(&self.message, &metadata)
+                .await
+            {
+                return self.on_reprepare::<String>(None, ReprepareReason::RevertedOrReorged);
```

Comment:
should this be `ErrorEstimatingGas`? as the doc for RevertedOrReorged seems like it relates to the tx actually landing on chain at some point `/// Delivery transaction reverted or reorged`